app/helpers/application_helper.rb

```diff
@@ -44,12 +44,28 @@ def abyme_for(association, position = :after, &block)
     end
   end
 
-  def abyme_add_association(text, **options)
-    button_tag(text, { data: { action: 'click->abyme#add_association' } }.merge(options))
+  def abyme_add_association(options = {}, &block)
+    action = 'click->abyme#add_association'
+    options[:attributes] = {} if options[:attributes].nil?
+    create_button(action, options, &block)
   end
 
-  def abyme_remove_association(text, **options)
-    button_tag(text, { data: { action: 'click->abyme#remove_association' } }.merge(options))
+  def abyme_remove_association(options = {}, &block)
+    action = 'click->abyme#remove_association'
+    options[:attributes] = {} if options[:attributes].nil?
+    create_button(action, options, &block)
+  end
+
+  private
+
+  def create_button(action, options, &block)
+    if block_given?
+      button_tag({data: { action: action }}.merge(options[:attributes])) do
```

Comment:
A mon avis, vaut mieux mettre une div simple et pas forcer le fait d'utiliser un bouton. Comme ça notre wrapper est un peu plus agnostique qu'un bouton qui est quand même plus contraignant je trouve...

Reply:
je pense que le tag par default doit rester un button c'est le comportement qu'on va vouloir dans la plupart des cas mais t'as raison c’était trop rigide j'ai modifié la method create_button pour qu'on puisse passer une clé tag et donc creer l'element html qu'on veut. le default reste button si on précise pas de tag

Reply:
You the best